spec/plaindatetime.html

```diff
@@ -979,23 +982,23 @@ <h1>ToTemporalDateTime ( _item_ [ , _options_ ] )</h1>
       <emu-alg>
         1. If _options_ is not present, set _options_ to *undefined*.
         1. Assert: Type(_options_) is Object or Undefined.
+        1. Let _resolvedOptions_ be ? SnapshotOwnProperties(? GetOptionsObject(_options_), *null*).
```

Comment:
Do we know the full set of options properties that we may need?  

If yes, then why don't we extract them here instead of enumerating all properties?

If no, is the reason that CalendarFields or some other custom calendar method can use custom options?  If so, then would using SnapshotOwnProperties break the case of using non-enumerable properties, non-own properties, etc.?

Reply:
See #2447, https://github.com/tc39/proposal-temporal/issues/2289#issuecomment-1343182389 for the context of when SnapshotOwnProperties was introduced. The tl;dr is that we clone the options object before sending it to a custom calendar method.

I'm not concerned about non-own properties. That almost certainly indicates either a bug (you have an unexpected `overflow` property in your options bag's prototype chain that you didn't intend to be passed to the function) or malicious code (e.g. some code sets `Object.prototype.overflow = 'reject'` to change the default behaviour of Temporal for other libraries.)

I'm also not concerned about non-enumerable own properties, since you really have to go out of your way to define such a property on an options object.